skgstat/MetricSpace.py

```diff
@@ -243,3 +243,70 @@ def dists(self):
 
         # return
         return self._dists
+
+class ProbabalisticMetricSpace(MetricSpace):
+    """Like MetricSpace but samples the distance pairs only returning a
+       `samples` sized subset. `samples` can either be a fraction of
+       the total number of pairs (float < 1), or an integer count.
+    """
+    
+    def __init__(self, coords, dist_metric="euclidean", max_dist=None, samples=0.5):
+        self.coords = coords.copy()
```

Comment:
I would add a docstring here, as the user should be able to directly interact with this class, right?